gradle-jdks/src/main/java/com/palantir/gradle/jdks/AzulZuluJdkDistribution.java

```diff
@@ -66,6 +69,18 @@ private static String arch(Arch arch) {
         throw new UnsupportedOperationException("Case " + arch + " not implemented");
     }
 
+    private static Extension extension(Os operatingSystem) {
+        switch (operatingSystem) {
+            case MACOS:
+                // fall-through
+            case LINUX:
+                return Extension.TARGZ;
+            case WINDOWS:
+                return Extension.ZIP;
+        }
+        throw new IllegalArgumentException("Unknown OS: " + operatingSystem);
```

Comment:
nit: We use `UnsupportedOperationException` above but `IllegalArgumentException` here. I don't have a preference, but we should be consistent.

Reply:
```suggestion
        throw new UnsupportedOperationException("Unknown OS: " + operatingSystem);
```
Good call, thanks!